pkg/cmd/roachtest/operations/changefeeds/create.go

```diff
@@ -0,0 +1,166 @@
+	options = append(options, sinkOptions...)
+	// Calculate whether the new changefeed should include an initial scan.
+	scanOption, err := calculateScanOption(allCFJobs)
+	if err != nil {
+		return err
+	}
+	options = append(options, scanOption)
+
+	o.Status(fmt.Sprintf("creating changefeed job to sink %s with options %v", sink, options))
+
+	// Construct and execute the SQL statement to create the changefeed.
+	_, err = conn.ExecContext(ctx, fmt.Sprintf("CREATE CHANGEFEED FOR TABLE %s.%s INTO '%s' WITH %s;",
+		dbName, tableName, sink, strings.Join(options, ",")))
+	return err // Return nil if successful, or error if failed.
+}
+
+// getSinkConfigs returns the sink uri along with the options for creating the changefeed.
+// this will be extended later for more sinks
+func getSinkConfigs(_ context.Context, _ []*jobDetails) (string, []string, error) {
+	return "null://", make([]string, 0), nil
+}
+
+// calculateScanOption determines whether the new changefeed should have an initial scan based on existing jobs.
+func calculateScanOption(allCFJobs []*jobDetails) (string, error) {
+	yesCountPercent := 0
+	if len(allCFJobs) > 0 {
+		scanYesCount := 0
+		// Count the number of jobs that have initial scan set to 'yes'.
+		for _, j := range allCFJobs {
+			if v, ok := j.payload.options[changefeedbase.OptInitialScan]; ok {
+				if v == "yes" {
+					scanYesCount++
+				}
+			}
+		}
+
+		// Calculate the percentage of jobs with initial scan 'yes'.
+		yesCountPercent = scanYesCount * 100 / len(allCFJobs)
+	}
+
+	// Get the maximum percentage of changefeeds that can have initial scans enabled.
+	maxPctScanOn, err := helpers.EnvOrDefaultInt(maxPctChangeFeedsScanOn, defaultEnvValuesInt[maxPctChangeFeedsScanOn])
+	if err != nil {
+		return "", err // Return error if fetching environment variable fails.
+	}
+
+	// Randomly decide whether the new changefeed should have an initial scan based on the max percentage allowed.
+	r, _ := randutil.NewPseudoRand()
+	if yesCountPercent < maxPctScanOn && randutil.RandIntInRange(r, 0, 100) < maxPctScanOn {
+		return fmt.Sprintf("%s=yes", changefeedbase.OptInitialScan), nil // Return "yes" for initial scan.
+	}
+	return fmt.Sprintf("%s=no", changefeedbase.OptInitialScan), nil // Return "no" if the scan option is not enabled.
+}
+
+// enableRangeFeedIfNotDoneAlready checks if the rangefeed setting is enabled, and enables it if not.
+func enableRangeFeedIfNotDoneAlready(ctx context.Context, conn *gosql.DB) error {
+	// Query to check if rangefeed is enabled.
+	stmt := "SHOW CLUSTER SETTING kv.rangefeed.enabled"
+	rangefeedEnabled := false
+
+	// Execute query to check the current status of rangefeed.
+	err := helpers.ExecuteQuery(ctx, func(rowValues []string) error {
+		rangefeedEnabled = rowValues[0] == "true"
+		return nil
+	}, conn, stmt)
+
+	// If rangefeed is already enabled or an error occurs, return the error.
+	if rangefeedEnabled || err != nil {
+		return err
+	}
+
+	// Enable rangefeed if it is not already enabled.
```

Comment:
not sure if all this is necessary. setting it one while its already on is fine no?

Reply:
This was to ensure that we need not do anything outside of the operations. But, I agree as this is a unnecessary repeat. I have removed this code.